src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala

```diff
@@ -43,6 +46,71 @@ trait JsonMethodsImplicits {
       }
     } else Left(InvalidParams(s"Quantity '${input.s}' should have 0x prefix"))
   }
+
+  private def decode(s: String): Array[Byte] = {
+    val stripped = s.replaceFirst("^0x", "")
+    val normalized = if (stripped.length % 2 == 1) "0" + stripped else stripped
+    Hex.decode(normalized)
+  }
+
+  protected def extractAddress(input: String): Either[JsonRpcError, Address] =
+    Try(Address(input)).toEither.left.map(_ => InvalidAddress)
+
+  protected def extractAddress(input: JString): Either[JsonRpcError, Address] =
+    extractAddress(input.s)
+
+  protected def extractBytes(input: String): Either[JsonRpcError, ByteString] =
+    Try(ByteString(decode(input))).toEither.left.map(_ => InvalidParams())
+
+  protected def extractBytes(input: JString): Either[JsonRpcError, ByteString] =
+    extractBytes(input.s)
+
+  protected def extractQuantity(input: JValue): Either[JsonRpcError, BigInt] =
```

Comment:
Yeah, the other extractors didn't suit me (though the differences are small), so I created my own versions. I agree it's better we stick with one version, but I wanted to get this reviewed first.